### ISSUE TITLE
Recurrence rule in release is broken

### ISSUE THREAD

**Issue by JoeyStk** [digitalfabrik/integreat-cms#3022]
### Describe the Bug

The recurrence rule in #3018 is (still) broken. As long as this is not resolved, our release is blocked

### Steps to Reproduce

1. Log on to our test system, go to events
2. Create a new event, set as recurrence rule every last Thursday starting in August (every month, last week, Thursday, every 1st...)
3. Go to the web app
4. See that for every month the 29th is shown

### Expected Behavior

Show for an event that happens every last Thursday the list of last Thursdays (so August 29th, September 26th, October 31st, November 28th, etc.).

### Actual Behavior

 (1) The current state **on production** is that only August 29th, October 31st and January are shown (months that have a fifth week). (2) [#2992](https://github.com/digitalfabrik/integreat-cms/pull/2992) attempted to fix this, but **on the test system** now the 29th of a month is shown.

### Additional Information

![grafik](https://github.com/user-attachments/assets/fd879d2d-01b4-4f64-8401-d9f62bf48a97) State after #2992
![grafik](https://github.com/user-attachments/assets/2b92735b-50f7-4046-9e83-ac4bc0f82ccb) State on production, before #2992

**Comment by PeterNerlich** [comment]
On a side note, I think the way we implement and display monthly recurrence for a weekday is misleading. We display *First/Second/Third/Last Week*, but we actually *mean* the *n*th occurrence of that *weekday* in the month. *(I would expect the month to be segmented into weeks from weekend to weekend and the counting be based on that with what we current show – but the start of those equivalency groups is actually based on which weekday the month starts or ends on)*

**Comment by PeterNerlich** [comment]
Ah, #2992 misunderstood the original issue #2989 for an invalid rendering of the iCal format, a bug they likely determined to be the intended target because of the difficulties of identifying the part of the CMS where the event data is handed over to the app *(navigating from the function implementing the generation of event instances to our actual API endpoint exposing it proved confusing, even if it was not the worst one I encountered so far in this project)*.
I decided to change how we store and handle the intention of "last occurrence in month", rendering the functional changes of #2992 unnecessary, but it should be closer to how iCal does it and be less ambiguous *(using negative values to count from the end of the month instead of the start)*

**Comment by david-venhoff** [comment]
I think this issue is invalid:
The `test 5 Woche` event is set to reccur every 3rd week (contradicting the title), if this is the correct event that is shown in the screenshot: https://cms-test.integreat-app.de/muenchen/events/de/4599/edit/

However, that still does not explain the dates listed at https://webnext.integreat.app/muenchen/de/events/test-5-woche ^^

That appears to be an app issue though, because we pass the correct recurrence rule to it, as can be seen in the network manager in the browser devtools of the page. Here is the relevant snippet: 
![image](https://github.com/user-attachments/assets/3ccb1fb5-749a-41b9-b9d2-1d7e34ecf482)

As can be seen, the recurrence rule is correctly set to repeat every third week of the month.

Because the `combine_recurring` parameter is set, our `iter_after` function never gets called even if it is technically incorrect right now.

**Comment by PeterNerlich** [comment]
Are we positive that the app only uses the endpoint with `?combine_recurring=True`?

**Comment by LeandraH** [comment]
Yes, I could fix the issue in the frontend in my local build. I rewrote the parsing of the recurrence rule a couple of weeks ago to fix a different bug and seem to have introduced this one, sorry! I'm currently adding more tests and will open a PR (and an issue) in the app repo later today.

**Comment by LeandraH** [comment]
> Are we positive that the app only uses the endpoint with `?combine_recurring=True`?

Yes, at least currently :)

**Comment by PeterNerlich** [comment]
Moved to backlog since the API endpoint is unused anyway at the moment

**Comment by MizukiTemma** [comment]
Wait for #3847 

If the API endpoint is removed, close this issue.

**Comment by osmers** [comment]
I think it is solved anyways - at least I tested according to the steps to reproduce and didn't find an issue, So closing
